src/libraries/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs

```diff
@@ -11,6 +11,8 @@ internal static partial class GlobalizationMode
 
         internal static bool UseNls => false;
 
+        internal static bool PredefinedOnly { get; } = GetSwitchValue("System.Globalization.PredefinedOnly", "DOTNET_SYSTEM_GLOBALIZATION_PREDEFINED_ONLY");
```

Comment:
>PredefinedOnly [](start = 29, length = 14)

are you adding this for Linux only? looks the calling site will call it even running on Windows.

Reply:
This is for Browser only.

Reply:
Why we need to restrict this to the browser only? We may enable this flag globally if we need to. Note the calling site of this property is called from shared code which runs with ICU and Nls. Give me a few minutes and I can send you what I propose. 

Also, does the browser always runs with Icu?

Reply:
I guess I just assumed only Browser would need this use case, but I can definitely move to GlobalizationMode.cs. And yes, the browser always runs with icu.

Reply:
I would suggest to move the checks 
https://github.com/dotnet/runtime/blob/f1ad1b97e9387dc700bf72917f29fc7d1258ac1a/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.Nls.cs#L14
and 
https://github.com/dotnet/runtime/blob/f1ad1b97e9387dc700bf72917f29fc7d1258ac1a/src/libraries/System.Private.CoreLib/src/System/Globalization/CultureInfo.Icu.cs#L14

to its own methods, something like 

```C#
[MethodImpl(MethodImplOptions.AggressiveInlining)]
internal static IcuIsEnsurePredefinedLocaleName(string name) // may put this in CultureData.Icu.cs
{
        Debug.Assert(!GlobalizationMode.UseNls);

        if (!Interop.Globalization.IsPredefinedLocale(name))
       {
            throw new CultureNotFoundException(nameof(name), SR.Format(SR.Argument_InvalidPredefinedCultureName, name));
        }
}

[MethodImpl(MethodImplOptions.AggressiveInlining)]
internal static NlsIsEnsurePredefinedLocaleName(string name) // may put this in CultureData.Nls.cs
{
            Debug.Assert(GlobalizationMode.UseNls);

            if (CultureData.GetLocaleInfoExInt(name, Interop.Kernel32.LOCALE_ICONSTRUCTEDLOCALE) == 1)
            {
                throw new CultureNotFoundException(nameof(name), SR.Format(SR.Argument_InvalidPredefinedCultureName, name));
            }}

```

Then replace the code here with 

```C#
 if (GlobalizationMode.PredefinedOnly)
{
    if (GlobalizationMode.UseNls)
    {  
        NlsIsEnsurePredefinedLocaleName(cultureName);
    }
    else
    {  
        IcuIsEnsurePredefinedLocaleName(cultureName);
    }
}
````


Last, replace the checks in the links I mentioned above with the new methods.

Reply:
Ok yes that makes sense.